src/components/presentations/NewArticle/NewArticle.jsx

```diff
@@ -0,0 +1,277 @@
+import React, { Component } from 'react';
+import { Input, Button, Dropdown, TextArea } from 'semantic-ui-react';
+import {
+  Editor,
+  EditorState,
+  RichUtils,
+  AtomicBlockUtils,
+  convertToRaw,
+} from 'draft-js';
+import { stateToHTML } from 'draft-js-export-html';
+import PropTypes from 'prop-types';
+import uploadToCloudinary from '../../../utils/uploadToCloudnary';
+import './new-article.scss';
+import BlockStyleToolbar from './BlockStyles/BlockStyleToolbar';
+
+class NewArticle extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      data: { title: '', abstract: '', body: '', keywords: '', category: '' },
+      headerpic: '',
+      editorState: EditorState.createEmpty(),
+    };
+    this.onChange = editorState => {
+      const contentState = editorState.getCurrentContent();
+      const html = stateToHTML(contentState);
+      const converted = (convertToRaw(contentState), html);
+      this.setState({ body: converted, editorState });
+    };
+    this.toggleBlockType = this.toggleBlockType.bind(this);
```

Comment:
can we remove the `bind(this)`? i think we can still achieve this by just doing `this.toggleBlockType` or we can use an arrow function `{() => this.toggleBlockType}`

Reply:
Actually using the `bind` syntax is better. Using an inline arrow function creates a new function on every render which is not good for performance. The other option could be to use an arrow function declared as a class property @vic3king